lib/data/repositories/transaction_repository_impl.dart

```diff
@@ -27,6 +29,18 @@ class TransactionRepositoryImpl extends SyncEntityRepository<AppDatabase,
           syncHandler: syncHandler,
         );
 
+  void _syncWithErrorHandling(Future<dynamic> Function() syncOperation) {
```

Comment:
The errors that matters are caught at the level of the `drift_sync_core/lib/src/adapters/rest/rest_sync_type_handler.dart`, if we want catch any other and log to firebase distribution I will rather we do that at the level of `drift_sync_core/lib/src/sync_entity_repository.dart` so it is done just twice for `putRemote` and `deleteRemote` function. 
Otherwise this function that wrapps put and delete in the repository implementation will have to be repeated for every entity. If you notice this is done just for the transaction implications class